### PR TITLE
fix: Trim trailing non-word characters for hostname labels

### PR DESCRIPTION
**Pull request by mrseanbaines**
Fixes an issue where hostname labels end in non-word characters which prevents certain browsers (e.g. Firefox) from loading the URL

### CODE REVIEW

**Comment by mrseanbaines** [comment]
@macko911 is there a way to test this change in real life?

**Comment by macko911** [comment]
> is there a way to test this change in real life?

@mrseanbaines  It is possible but it's not trivial. I wanted to write a guide on how to do so, if you don't mind waiting until early next week I would take time today to finish it, then use you as a guinea pig to see if the guide works, wdyt? 😄 
https://www.notion.so/pleo/Guide-to-test-terraform-changes-before-merging-e639098417574a97ae180e9c9b7e644f

**Comment by mrseanbaines** [comment]
> > is there a way to test this change in real life?
> 
> @mrseanbaines It is possible but it's not trivial. I wanted to write a guide on how to do so, if you don't mind waiting until early next week I would take time today to finish it, then use you as a guinea pig to see if the guide works, wdyt? 😄 https://www.notion.so/pleo/Guide-to-test-terraform-changes-before-merging-e639098417574a97ae180e9c9b7e644f

Ha, yep that works for me 👍

File: actions/cursor-deploy/index.ts [hunk 1]
```diff
@@ -153,6 +153,7 @@ export function branchNameToHostnameLabel(ref: string) {
         .replace(/-{2,}/gi, '-') // get rid of multiple consecutive "-"
         .toLowerCase()
         .slice(0, 60)
+        ?.replace(/[^\w]$/gi, '') // get rid of trailing non-word characters
```
Comment (binaryannie) [inline actions/cursor-deploy/index.ts]:
```suggestion
        ?.replace(/-$/gi, '') // get rid of trailing "-", it breaks firefox
```

**Review by binaryannie — APPROVED** [review]

File: actions/cursor-deploy/index.ts (hunk 1 above)
Comment (mrseanbaines) [reply actions/cursor-deploy/index.ts]: https://github.com/pleo-io/spa-tools/pull/200/commits/cb7f143e9ff9bfc4879b68ec38716ef8fc326e3f

Left out the explanation part as none of the other transformations have explanations. Also I don't know if it's unique to Firefox

**Comment by macko911** [comment]
> > > is there a way to test this change in real life?
> > 
> > 
> > @mrseanbaines It is possible but it's not trivial. I wanted to write a guide on how to do so, if you don't mind waiting until early next week I would take time today to finish it, then use you as a guinea pig to see if the guide works, wdyt? 😄 https://www.notion.so/pleo/Guide-to-test-terraform-changes-before-merging-e639098417574a97ae180e9c9b7e644f
> 
> Ha, yep that works for me 👍

[The guide](https://www.notion.so/pleo/Guide-to-testing-Terraform-changes-before-merging-using-env0-e639098417574a97ae180e9c9b7e644f) should be ready, feel free to add comments when something is unclear, it's pretty late so I'm not really sure it makes sense 😄

**Review by macko911 — APPROVED** [review]

**Pull request merged**